eval.go

```diff
@@ -318,7 +322,12 @@ nextFileMatch:
 		}
 
 		if opts.UseBM25Scoring {
-			d.scoreFileUsingBM25(&fileMatch, nextDoc, finalCands, opts)
+			// For BM25 scoring, the calculation of the score is split in two parts. Here we
+			// calculate the term frequencies for the current document and update the
+			// document frequencies. Since we don't store document frequencies in the index,
+			// we have to defer the calculation of the final BM25 score to after the whole
+			// shard has been processed.
+			calculateTermFrequency(&fileMatch, finalCands, df)
```

Comment:
can't we make another slice that has the invariant that `scoreData[i] belongs to res.Files[i]`. Adding private fields to our public API feels very wrong.

Reply:
you prefer [af9a0a4](https://github.com/sourcegraph/zoekt/pull/788/commits/af9a0a4bc861750531f2ee90ddac53feaedeb7cc)? 

>> Adding private fields to our public API feels very wrong.

We probably should have internal version of our structs and convert at the API boundary. I don't feel that strongly about adding private fields though if it helps to keep the code simple.

Reply:
I reverted the changes to the API/`FileMatch` and went back to something similar to the first version, but with the BM25 calculation all in one function. I think it is a good trade-off. A ran the evals again without change.